### ISSUE TITLE
spurious warnings about question or exclamation marks in a header

### ISSUE THREAD

**Issue by wbolster** [DavidAnson/markdownlint#58]
a header like this one:

``` markdown
# Why?
```

produces a `MD026` warning about trailing punctutation in the header. the same happens when an exclamation mark is used.

this should not happen.

**Comment by DavidAnson** [comment]
What do you think is wrong here?

The documentation for `MD026` explicitly calls out that it will trigger for `?` and `!`:
https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md026---trailing-punctuation-in-header

If you do not like this rule, it can be disabled completely - or you can edit the list of characters it looks for (again, see documentation above).

**Comment by wbolster** [comment]
what i think is wrong here is that the defaults are not sensible enough. :)

having a question mark at the end of a header is quite common (e.g. faqs), and requiring explicit configuration for it seems backwards to me.

**Comment by DavidAnson** [comment]
Fair enough. :) For what it's worth, this rule came from the Ruby implementation. I'll admit it's not my absolute favorite rule, but I think it's a good guideline in general, it's easy to turn off if you don't want it, and it's easy to customize when needed. I'm not inclined to break compatibility with the Ruby implementation over something like this, so unless you find that the rule is behaving wrongly, I will recommend turning it off. Sorry for the inconvenience!

**Comment by wbolster** [comment]
yeah i understand you took the rules from another project (unsure why there are 2 implementations though), but i was not aware that you aim for keeping those rules unchanged, even when they do not make sense as defaults.

thanks for the explanation.

**Comment by petk** [comment]
Hello, worth noting, that headings in writings can include question marks an exclamation marks. Yes, one of the more strange settings.
